dao/src/main/java/com/epam/esm/dao/impl/CertificateTagsDaoImpl.java

```diff
@@ -14,23 +16,26 @@
 @Repository
 public class CertificateTagsDaoImpl implements CertificateTagsDao {
 
-    private final JdbcTemplate template;
+    private final EntityManager entityManager;
 
     @Autowired
-    public CertificateTagsDaoImpl(JdbcTemplate template) {
-        this.template = template;
+    public CertificateTagsDaoImpl(EntityManager entityManager) {
+        this.entityManager = entityManager;
     }
 
     /**
-     * This method saves link between GiftCertificate entity and Tag entity.
+     * This method saves a link between GiftCertificate entity and Tag entity.
      *
      * @param certificateId id of GiftCertificate entity
      * @param tagId         id of Tag entity
      * @since 1.0
      */
     @Override
     public void save(long certificateId, long tagId) {
-        template.update(ApplicationConstants.SAVE_CERTIFICATE_TAGS_QUERY, certificateId, tagId);
+        Query nativeQuery = entityManager.createNativeQuery(ApplicationConstants.SAVE_CERTIFICATE_TAGS_QUERY);
```

Comment:
For what reason do you use native query?
Is it possible to use CriteriaBuilder here?

Reply:
This method inserts data in relation table between GiftCertificate entity and Tag entity. I can't use criteria api and HQL here because I dont have any mapped entity to work with this table.

Reply:
Ok